### ISSUE TITLE
Docstring as alternative description for test method/class

### ISSUE THREAD

**Issue by alexandermomot** [pytest-dev/pytest-html#34]
One more suggest from me. Hope I'm not impudent, well not yet :)

Docstring is a cool feature, and with it we don't have to describe out tests somewhere else istead of test itself. I would suggest to read docstring from class/function and append it to the title here:
https://github.com/davehunt/pytest-html/blob/master/pytest_html/plugin.py#L154
optionally. Optionally, because when someone wants to perform doctest (dunno if there someone which does :) ), the feature will destroy the report.

**Comment by davehunt** [comment]
Thanks for all the suggestions! Perhaps the description could be shown when you hover over the test name to prevent excessively long names?

**Comment by davehunt** [comment]
@alexandermomot I've fixed this by allowing users to implement their own hooks to modify the results table, and given an example that adds a description column with the docstring of the test function. If you have some time, please take a look over #104 and leave feedback. Hopefully I can release 1.14.0 soon with this enhancement.

**Comment by davehunt** [comment]
Fixed by adding hooks in bc2e8a3 available in v1.14.0.

**Comment by MartykQ** [comment]
In `conftest.py`:


```
import pytest
from py.xml import html

def pytest_html_results_table_header(cells):
    cells.insert(2, html.th('Description'))

def pytest_html_results_table_row(report, cells):
    cells.insert(2, html.td(report.description))

@pytest.hookimpl(hookwrapper=True)
def pytest_runtest_makereport(item, call):
    outcome = yield
    report = outcome.get_result()
    report.description = str(item.function.__doc__)

```
